# generated by patch-package
#
# declared package:
#   @tanstack/query-persist-client-core: 4.13.0
#
diff --git a/node_modules/@tanstack/query-persist-client-core/build/lib/persist.esm.js b/node_modules/@tanstack/query-persist-client-core/build/lib/persist.esm.js
index 12a3412..fcbec93 100644
--- a/node_modules/@tanstack/query-persist-client-core/build/lib/persist.esm.js
+++ b/node_modules/@tanstack/query-persist-client-core/build/lib/persist.esm.js
@@ -13,6 +13,10 @@ async function persistQueryClientRestore({
   buster = '',
   hydrateOptions
 }) {
+  // pause all queries until queryClient is restored
+  // this is used in node_modules/@tanstack/query-core/build/lib/queryObserver.esm.js
+  queryClient.paused = true
+
   try {
     const persistedClient = await persister.restoreClient();
 
@@ -38,6 +42,14 @@ async function persistQueryClientRestore({
 
     persister.removeClient();
   }
+
+  // queryClient is restored -> restart paused queries
+  queryClient.paused = false
+  if (queryClient.afterPersistQueryClientRestore) {
+    for (const restartQuery of queryClient.afterPersistQueryClientRestore) {
+      restartQuery()
+    }
+  }
 }
 /**
  * Persists data from the QueryCache
